File: python/p5.py
import operator as op

r = range(2, 21)

def get_factors(num):
    rslt = dict((x, 0) for x in r)
    n = num
    i = 2
    while n > 1 and i <= 20:
        while n % i == 0:
            n /= i
            rslt[i] += 1
        i += 1
    return rslt

all_factors = [get_factors(x) for x in r]
factors = dict((i, max(d[i] for d in all_factors)) for i in r)
rslt = reduce(op.mul, [k**v for k, v in factors.items() if v > 0])
print rslt

File: js/p5.js
var _ = require('underscore');

_.mixin({
    makedict : function(keys, valfunc) {
        return _.object(keys, _.map(keys, valfunc));
    },

    product : function(list) {
        return _.reduce(list, function(x, y){ return x * y; }, 1);
    }
});


var r = _.range(2, 21);

var get_factors = function(n) {
    var rslt = _.makedict(r, function(i){ return 0; });
    for (var i = 2; i <= 20 && n > 1; i++) {
        while (n % i == 0) {
            n /= i;
            rslt[i]++;
        }
    }
    return rslt;
};

var factors = _.makedict(r, function(i) {
    var all_factors = _.map(r, get_factors);
    return _.max(_.pluck(all_factors, i));
});

factors = _.map(_.pairs(factors), function(li) {
    var k = li[0], v = li[1];
    return Math.pow(k, v);
});

var rslt = _.product(factors);

console.log(rslt)
